### DISCUSSION TITLE
Infrared using ROS2

### DISCUSSION THREAD

**Discussion by fmorshed52** [iRobotEducation/create3_docs#373] [Questions]
Has anyone programmed an infrared detector using ROS 2 for create 3?  If so, do you mind to sharing your experience?

I am trying to make the create 3 robot to follow a 3 inch wide white plastic line.  I think infrared devices are what I need.  Searching the net I could not find anything specific for ROS 2 and create 3 robot.  Although there are infrared devices available for sale.

**Comment by fmorshed52** [comment]
Can any of the existing sensors on create 3 be used as an Infrared sensor to follow a 3 inch wide white plastic line?  I have read that reflection of Infrared sensors can determine whether the reflection is from a pale or a darker (less reflective) surface.  This would help me follow the line.  I don't know whether I can use one of the existing sensors on create 3 or I need to buy them and add them.

Any ideas?

**Reply by shamlian** [reply]
It was requested [here](https://github.com/iRobotEducation/create3_docs/issues/320) for the analog values to be available from the robot's cliff sensors. This is on the roadmap. In the mean time, you could try to find a surface black enough to "trick" the cliff sensors and use the hazard output of the robot.

**Reply by fmorshed52** [reply]
Could you expand on what you mean by "find a surface black enough to "trick" the cliff sensors and use the hazard output of the robot" please?  I am not sure what you mean by "black enough".  Do you mean black surrounding the white plastic line in my case?

I read https://github.com/iRobotEducation/irobot_create_msgs/blob/rolling/msg/HazardDetection.msg.  I see the Cliff, type and header.

**Reply by justinIRBT** [reply]
Black enough meaning dark enough to absorb the IR light from the cliff sensor to make the robot think it is seeing a cliff (if you achieve this, you will additionally need to disable the cliff reflex).
